hw/ip/spi_device/doc/_index.md

```diff
@@ -62,6 +62,51 @@ The TPM submodule conforms to the [TPM over SPI 2.0][] specification. The TPM op
 
 ![Block Diagram](block_diagram.svg)
 
+In Generic mode, the incoming data is stored byte-based into an asynchronous FIFO.
+The logic inside the generic mode then updates the DPSRAM RX space.
+The logic also reads data from the DPSRAM then pushes out to the SPI MISO line.
+
+The Generic mode uses the entire DPSRAM space exclusively.
+The TX/RX size in the DPSRAM can be changed by compile-time parameters.
+
+When Flash mode is selected, the command parser accepts the first byte of the SPI MOSI line then activates the flash submodules, such as Status, JEDEC, Read command, and Upload function.
+The Status logic processes the three Read Status commands.
+The SW may configure three bytes of the Flash Status CSR then the Status submodule returns the CSR data into the SPI MISO line.
+The SW may configure the Read Status commands' opcodes.
+
+The JEDEC submodule returns the JEDEC Manufacturer ID followed by the additional information.
+The Manufacturer ID may vary depending on the company.
+For example, lowRISC JEDEC ID `EFh` follows twelve bytes of 7Fh Continuous Codes, requiring a total thirteen bytes for the manufacturer ID.
+The SW may configure how many Continuous Codes is needed and the actual manufacturer ID.
+
+The Read submodule processes the Read SFDP (Serial Flash Discoverable Parameters) command, and up to six different types of the read commands.
+The read submodule receives address information from the SPI transaction, fetches the data from the read buffer in the DPSRAM, and returns the data on SPI lines (single, dual, quad lines).
+If the received address falls into the SW programmable mailbox address space, the logic fetches data not from the read buffer but from the mailbox buffer in the DPSRAM.
+
+SW may configure command information slots to upload the command into the FIFOs and the payload buffer in the DPSRAM.
+SW may additionally let HW to set the BUSY bit in the Status register when the HW uploads the command.
+
+In Passthrough mode, the logic filters the incoming transaction if the transaction is not permitted.
+The SW may configure the logic to change a portion of the address or first 4 bytes of the payload.
+
+## Hardware Interfaces
+
+{{< incGenFromIpDesc "../data/spi_device.hjson" "hwcfg" >}}
+
+The TPM submodule requires a separate input port for CS#.
+The TPM submodule and other SPI Device modes are able to be active together.
+The SB distinguishes between the TPM transactions and the other SPI transactions using separate CS# ports.
```

Comment:
`SB`..?

Reply:
ah okay, early on in the documentation it says south bridge. 
is that terminology vendor specific..?

Reply:
OK it would be better to use the host system or SPI host system.